### ISSUE TITLE
Merge / Build problem

### ISSUE THREAD

**Issue by Tatsuya79** [libretro/mame#45]
I saw the update to something like Mame 0.171 in the repo but using mame_libretro.dll for win x64 from the buildbot it doesn't seem to be up to date.

There was a fix for games like akumajo on X68000 some weeks ago: 
http://mametesters.org/view.php?id=6139

Here it's still bugged like in 0.170.

**Comment by sergiobenrocha2** [comment]
It's not exactly 0.171, the repo was merged with the master branch a few days later the release. Time to merge the tags instead.

@r-type

**Comment by Tatsuya79** [comment]
I've been trying the core from the buildbot which should be 0.172 after today merge.
This bug is still there while it's been fixed shortly after 0.170 release in stand-alone Mame.

Did some commits got left out?

**Comment by r-type** [comment]
@Tatsuya79  can you provide an img to see where is the blue exactly ?

**Comment by Tatsuya79** [comment]
Just after the Konami logo and sometimes in the intro:
http://postimg.org/image/eynscbo8r/
Or you can skip the intro and wait a bit for stage 2 demo:
http://postimg.org/image/g27woa8vv/

It's a palette bug that has been corrected 3 months ago.
It happens with several X68000 games like "directx" right from the start (yes, that's a game).

**Comment by r-type** [comment]
strange , i don't have the pb with libretro mame 0.172 or current.
no blue colours like you.

**Comment by Tatsuya79** [comment]
Is it the core from the buildbot?

**Comment by r-type** [comment]
No i m not using buildbot and compile the core myself. And i m on linux.
Btw maybe you have an cfg file that overwriten
The default config.

**Comment by Tatsuya79** [comment]
I tried to remove everything: nvram, cfg, ini...
Still the same.
I don't think this palette bug can be caused by that as it was fixed in source.

I'm using the buildbot core for win x64:
http://buildbot.libretro.com/nightly/windows/x86_64/latest/mame_libretro.dll.zip

**Comment by r-type** [comment]
tested with the core you link
using wine  and got the blue issue like you.
BTW  the core indicate 0.170 in the menu , my own linux build it indicate 0.172.

**Comment by Tatsuya79** [comment]
Thanks for your confirmation.
So the buildbot fails and fall back to an old build for win x64?

**Comment by fr500** [comment]
if the buildbot fails it just fails, no fallback it leaves the old core there.

**Comment by Tatsuya79** [comment]
Trying to build it with that environment https://gist.github.com/heuripedes/b6da0fea68b648bdd2ae
Is the command still like this? with Python 2.7?
make -f Makefile.libretro PTR64=1 PYTHON="C:/Progra~1/Python/pythonw.exe" -j4

**Comment by twinaphex** [comment]
Instead of wasting more time on an already obsolete version (0.172), I updated it to mainline again.

I am currently stuck on this code -

http://hastebin.com/opitefutuv.vala

let's make this work now and let's not fall disastrously behind again like before. This time I will take a more proactive approach in this repo to make sure that doesn't happen again, it happened one too many times already before.

**Comment by r-type** [comment]
> Instead of wasting more time on an already obsolete version (0.172), I updated it to mainline again.

Lol ,no wast here for me , BTW  i updated mine 2 day ago and it working fine !

https://github.com/r-type/mame

**Comment by twinaphex** [comment]
Yeah I did a git pull right now of your repo.

However I am still stuck at the same location. Know what could be up?

**Comment by sergiobenrocha2** [comment]
Could we merge the tags instead master branch, in the next releases?

```git fetch --tags origin && git merge mame0174```

**Comment by twinaphex** [comment]
I'm currently compiling r-type's repo, if that works, I'll just kill this one and fork from his repo instead. That'll save us spending time on getting this to actually work again.

**Comment by twinaphex** [comment]
OK, r-type's repo worked. Moving this one over to my repo and then re-forking r-type's repo again on the libretro org.

**Comment by r-type** [comment]
@twinaphex @fr500

can you explain why 
linux
http://buildbot.libretro.com/nightly/linux/x86_64/latest/
mame_libretro give me a 0.171 in rgui

windows (with wine)
http://buildbot.libretro.com/nightly/windows/x86_64/latest/
mame_libretro give me a 0.170 in rgui

and my own build based on last https://github.com/libretro/mame
give me mame_libretro give me a 0.174 in rgui

does build bot use outdated src ? it should be 0.172 before the last rebase on win/lin ?

**Comment by r-type** [comment]
@Tatsuya79 

i don't have the pb with 0.174 core , i bet 0.74 win core  will be avaible in updater soon  ..
but you can test a preview here 

http://dl.free.fr/getfile.pl?file=/BFMDGiEp


@fr500 if you catch the comment on the libco and seh , can you show me what TC i should use to avoid crash vith libco (on win) .

@twinaphex 

 btw , one day , we really should redone the libretro osd part . as the new menu ods is not very usable in this state !

**Comment by Tatsuya79** [comment]
Thanks that's appreciated.
Will test it after I put everything back together, just lost my main hard drive.

**Comment by fr500** [comment]
@r-type when it fails it leaves the latest core that built there.
So, mame hasn't built since 0.170 I guess on windows. I think it needs newer GCC than what we have on the windows server.

Regarding the toolchain, official mingw64 SJLJ

**Comment by r-type** [comment]
@fr500 ok that explain why @Tatsuya79  had pb with x68000,
  thanks , I'll fix my  x86_64-w64-mingw32- toolchain to be sjlj instead of seh.

@twinaphex , then you should remove the bad hack i've done in libco.c in your fork.
else it ll break your build.

**Comment by twinaphex** [comment]
OK, and yeah I agree we should redo the port in some areas and also look at getting bgfx working too for when the time comes when upstream MAME will switch to that.

**Comment by r-type** [comment]
Yeah for bgfx but we need to alt the bgfx upstream as in source they assume sdl or win( last time i ve checked)

**Comment by Tatsuya79** [comment]
@r-type 
The X68000 palette bug is gone with your version, but the ratio is wrong.
Akumajo X68000:
http://postimg.org/image/ck59nz5tn/
Arcade, 1941:
http://postimg.org/image/9os6h41tn/

Those should be full screen like my previous screenshot.

**Comment by r-type** [comment]
Yeah @sergiobenrocha2 point this issue after 0.172 update (on sfa2 )
I dont know why for now. But it s also why we should rework the osd.

**Comment by r-type** [comment]
Related to  #48
